src/lightning/pytorch/loggers/utilities.py

```diff
@@ -52,3 +55,42 @@ def _scan_checkpoints(checkpoint_callback: Checkpoint, logged_model_time: dict)
     )
     checkpoints = [c for c in checkpoints if c[1] not in logged_model_time.keys() or logged_model_time[c[1]] < c[0]]
     return checkpoints
+
+
+def _log_hyperparams(trainer: "pl.Trainer") -> None:
+    if not trainer.loggers:
+        return
+
+    pl_module = trainer.lightning_module
+    datamodule_log_hyperparams = trainer.datamodule._log_hyperparams if trainer.datamodule is not None else False
```

Comment:
datamodule is not required to have `_log_hyperparams`, so I think this needs to check if `_log_hyperparams` is an attribute of `trainer.datamodule`. @carmocca @Borda

Reply:
oops please disregard! I forgot to call `super().__init__()` in my subclass of pl.LightningDataModule